layers/+misc/all-the-icons/packages.el

```diff
@@ -0,0 +1,135 @@
+    ;; Notmuch
+    (add-to-list 'all-the-icons-mode-icon-alist
+                  '(notmuch-show-mode all-the-icons-octicon "mail-read" :face all-the-icons-lblue))
+    (add-to-list 'all-the-icons-mode-icon-alist
+                  '(notmuch-search-mode all-the-icons-material "inbox" :face all-the-icons-blue))
+    (add-to-list 'all-the-icons-mode-icon-alist
+                  '(notmuch-message-mode all-the-icons-material "email" :face all-the-icons-orange))
+
+    ;; Missing directory type (for ivy)
+    (add-to-list 'all-the-icons-icon-alist '("/$" all-the-icons-octicon "file-directory"))
+
+    ;; Switch to a go icon that shows up. The default one is practically invisible.
+    (add-to-list 'all-the-icons-mode-icon-alist '(go-mode all-the-icons-fileicon "go" :face all-the-icons-lblue))
+    (add-to-list 'all-the-icons-icon-alist '("\\.go$" all-the-icons-fileicon "go" :face all-the-icons-lblue))
+
+    ;; systemd
+    (add-to-list 'all-the-icons-mode-icon-alist '(systemd-mode all-the-icons-faicon "cogs" :face all-the-icons-dred))
+    (add-to-list 'all-the-icons-icon-alist `(,systemd-autoload-regexp all-the-icons-faicon "cogs" :face all-the-icons-dred))
+
+    ;; matrix
+    (add-to-list 'all-the-icons-mode-icon-alist '(matrix-client-mode all-the-icons-material "chat" :face all-the-icons-dcyan))
+
+    ;; Spacemacs home
+    (add-to-list 'all-the-icons-mode-icon-alist
+                  '(spacemacs-buffer-mode all-the-icons-material "home" :face all-the-icons-purple))))
+
+(defun all-the-icons/init-all-the-icons-dired ()
+    (use-package all-the-icons-dired
+      :defer t
+      :init
+      (add-hook 'dired-mode-hook 'all-the-icons-dired-mode)))
+
+(defun all-the-icons/init-all-the-icons-ibuffer ()
+  (use-package all-the-icons-ibuffer
+    :defer t
+    :init
+    (add-hook 'ibuffer-mode-hook 'all-the-icons-ibuffer-mode)))
+
+(defun all-the-icons/init-all-the-icons-ivy ()
+    (use-package all-the-icons-ivy
+      :after ivy
+      :config
+      (all-the-icons-ivy-setup)))
+
+(defun all-the-icons/post-init-ivy-rich ()
+  (with-eval-after-load 'ivy-rich
+    (defun ivy-rich-file-icon (candidate)
+      (let ((icon (all-the-icons-icon-for-file candidate)))
+        (if (symbolp icon)
+            (all-the-icons-icon-for-mode 'fundamental-mode)
+          icon)))
+    (defun ivy-rich-buffer-icon (candidate)
+      (let ((icon (all-the-icons-icon-for-mode
+                   (buffer-local-value 'major-mode (get-buffer candidate)))))
+        (if (symbolp icon)
+            (all-the-icons-icon-for-mode 'fundamental-mode)
+          icon)))))
+
+(defun all-the-icons/post-init-company-box ()
+  (setq company-box-icons-alist 'company-box-icons-all-the-icons)
+  (with-eval-after-load 'company-box
+    ;; Icons selected by liguangsheng
+    ;; https://github.com/liguangsheng/emacsd/blob/master/lisp/init-completion.el
+      (defun spacemacs/company-box-icon (family icon &rest args)
```

Comment:
Renamed from `my-company-box-icon`.

Reply:
Should be ok, company box is not working for me either with or without your layer, I have asked the others if it is working for either of them before we merge.

Reply:
Needs testing.